src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -306,6 +322,24 @@ private void fireNextRequest() {
         PatternsCS.pipe(responsePromise, context().dispatcher()).to(self());
     }
 
+    private CompletableFuture<Response> sendHttpRequest(
+            final Request request,
+            final int attempt) {
+        final CompletableFuture<Response> promise = new CompletableFuture<>();
+        _client.executeRequest(request, new ResponseAsyncCompletionHandler(promise));
+        promise.handle((result, err) -> {
```

Comment:
We're probably going to need a bit of a backoff here.  I would recommend using an exponential backoff strategy

Reply:
At least adding some support for backoff would be nice. But I'm also okay with adding that in a second step.